### PR TITLE
done

### PR DESCRIPTION
**Pull request by Oleksii-co**

### CODE REVIEW

File: routes/api/contacts.js [hunk 1]
```diff
@@ -1,25 +1,90 @@
...
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+import contactsService from "../../models/contacts.js";
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const contactAddSchema = Joi.object({
+  name: Joi.string().required(),
```
Comment (ceREAL133) [inline routes/api/contacts.js:11]: schema should be in separate file, you should validate email and phone properly

File: routes/api/contacts.js [hunk 2]
```diff
@@ -1,25 +1,90 @@
...
+  email: Joi.string().required(),
+  phone: Joi.string().required(),
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
```
Comment (ceREAL133) [inline routes/api/contacts.js:17]: functions in router should be only invoked, not defined, separate them on another file

File: routes/api/contacts.js [hunk 3]
```diff
@@ -1,25 +1,90 @@
...
+    const { contactId } = req.params;
+    const result = await contactsService.getContactById(contactId);
+    if (!result) {
+      throw HttpError(404, `Contact with id=${contactId} not found`);
+    }
+    res.json(result);
+  } catch (error) {
+    next(error);
+    //   const { status = 500, message = "Server error" } = error;
+    //   res.status(status).json({
```
Comment (ceREAL133) [inline routes/api/contacts.js:38]: delete comments

**Review by ceREAL133 — CHANGES_REQUESTED** [review]
PR should go into your repo, not in goitacademy:master

File: schemas/contact.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Joi from "joi";
+
+const addSchema = Joi.object({
+  name: Joi.string().required(),
```
Comment (ceREAL133) [inline schemas/contact.js:4]: add message to required

**Review by ceREAL133 — CHANGES_REQUESTED** [review]

File: schemas/contact.js (hunk 4 above)
Comment (Oleksii-co) [reply schemas/contact.js:4]: Не зрозумів, Що потрібно добавити?

**Pull request closed without merging**